src/inc/corinfo.h

```diff
@@ -1926,6 +1926,23 @@ struct CORINFO_VarArgInfo
 
 #include <poppack.h>
 
+#define SIZEOF__CORINFO_Object                            TARGET_POINTER_SIZE /* methTable */
+
+#define OFFSETOF__CORINFO_Array__length                   SIZEOF__CORINFO_Object
+#if defined(_TARGET_AMD64_) || defined(_TARGET_ARM64_)
+#define OFFSETOF__CORINFO_Array__u1Elems                  (OFFSETOF__CORINFO_Array__length + sizeof(unsigned) /* length */ + sizeof(unsigned) /* alignpad */)
+#else
+#define OFFSETOF__CORINFO_Array__u1Elems                  (OFFSETOF__CORINFO_Array__length + sizeof(unsigned) /* length */)
+#endif // defined(_TARGET_AMD64_) || defined(_TARGET_ARM64_)
+
+#define OFFSETOF__CORINFO_RefArray__length                SIZEOF__CORINFO_Object
+
+#define OFFSETOF__CORINFO_RefAny__dataPtr                 SIZEOF__CORINFO_Object
```

Comment:
Nit: It may be nice to call this `...TypedReference__dataPtr` to make this move obvious; or at least add a comment that says that it is layout of TypedReference type.